test/functional/terminal/buffer_spec.lua

```diff
@@ -155,5 +157,42 @@ describe('terminal buffer', function()
       :bnext                                            |
     ]])
   end)
+
+  it('handles loss of focus gracefully', function()
+    -- Temporarily change the statusline to avoid printing the file name, which
+    -- varies be where the test is run.
+    execute('set statusline=+++')
+    execute('set laststatus=0')
+
+    -- Save the buffer number of the terminal for later testing.
+    local tbuf = eval('bufnr("%")')
+
+    source([[
+    function! SplitWindow()
+      new
+      set statusline=+++
+    endfunction
+
+    startinsert
+    call jobstart(['sh', '-c', 'exit'], {'on_exit': function("SplitWindow")})
+    ]])
+
+    -- We should be in a new buffer now.
+    screen:expect([[
+      ^                                                  |
+      ~                                                 |
+      +++                                               |
+      rows: 2, cols: 50++                               |
+      {2: }                                                 |
+      {1:+++                                               }|
+                                                        |
+    ]])
```

Comment:
One outstanding issue: event when I put `execute('set noruler')`, that `rows: 2, cols: 50` shows up, and if the `statusline` is longer than one character, all but the first get appended after the `cols: 50`.

Otherwise, this is basically the output I expect.